kubermatic/kubermatic
# Add endpoint for OIDC kubeconfig secret for WEB terminal

**What does this PR do / Why do we need it**: add an endpoint for OIDC kubeconfig secret for the WEB terminal.

**Does this PR close any issues?**:
Fixes #

**Special notes for your reviewer**:
In order to test this locally:
1. Alter our dex configuration (`kubermaticIssuer` client) to whitelist the `redirectURI`: `http://localhost:8080/api/v2/kubeconfig/secret` and `https://kubermatic.io/api/v2/kubeconfig/secret`
2.
  - `kubectl -n oauth edit configmap dex`
  - `kubectl -n oauth rollout restart deployment dex`
![image](https://user-images.githubusercontent.com/2285385/173568073-c5a97665-28e0-4d55-80c6-f5133a091a12.png)
3. Create and wait for any cluster to start (using our dev env).
4. Run the API locally, i.e. `make clean && FEATURE_GATES=OIDCKubeCfgEndpoint=true KUBERMATIC_EDITION=ee KUBERMATIC_DEBUG=false ./hack/run-api.sh`
5. 
  Open WEB browser and enter: `http://localhost:8080/api/v2/kubeconfig/secret?cluster_id=<clusterID>&project_id=<projectID>&user_id=<userID>`. Make sure to replace `clusterID`,  `userID` and `projectID` with the values of the created cluster. The userID is the ID for the user object in the KKP.

6. Get user cluster kubeconfig and check if the secret was created in the `kube-system` namespace. The secret name is your email encoded to md5 hash. You can check the name here https://www.md5hashgenerator.com/. For example for user `lukasz.zajaczkowski@kubermatic.com` we will get `40ce6d00769edaab4eb1d8cacfea5cb1`

```release-note
Add an endpoint for OIDC kubeconfig secret for the WEB terminal
```
